src/pycsou/opt/solver/cg.py

```diff
@@ -0,0 +1,142 @@
+import warnings
+
+import pycsou.abc.operator as pyco
+import pycsou.abc.solver as pycs
+import pycsou.opt.stop as pycos
+import pycsou.runtime as pycrt
+import pycsou.util as pycu
+import pycsou.util.ptype as pyct
+
+
+def is_broadcastable(arr1, arr2):
+    shp1, shp2 = arr1.shape, arr2.shape
+    if shp1[-1] != shp2[-1]:
+        return False
+    for a, b in zip(shp1[:-1], shp2[:-1]):
+        if a == 1 or b == 1 or a == b:
+            pass
+        else:
+            return False
+    return True
+
+
+class CG(pycs.Solver):
+    r"""
+    Conjugate Gradient Method.
+    The Conjugate Gradient method solves the system of linear equations of the form
+    .. math::
+       {\mathbf{A}\mathbf{x} = \mathbf{b}},
+    for the vector :math:`\mathcal{x}: \mathbb{x}^N`.
+    where:
+    * :math:`\mathcal{A}: \mathbb{R}^M\times^N` is a *symmetric* *positive definite* matrix.
+    * :math:`\mathcal{b}: \mathbb{b}^M`.
+    """
+
+    def __init__(
+        self,
+        a: pyco.PosDefOp,
+        b: pyct.NDArray,
+        log_var: pyct.VarName = ("x",),
+    ):
+        super().__init__(
+            log_var=log_var,
+        )
+
+        self._a = a
+        self._b = pycrt.coerce(b)
+
+    def fit(
+        self,
+        x0: pyct.NDArray = None,
+        stop_crit: pycs.StoppingCriterion = None,
```

Comment:
If `tol` is provided in `fit()`, then you should not have `stop_crit` in the signature anymore.
It is the job of `fit()` to construct the AbsErr stop_crit and call `super().fit()` with it.